data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/validator/PostBlock.java

```diff
@@ -0,0 +1,115 @@
+import tech.pegasys.artemis.api.ValidatorDataProvider;
+import tech.pegasys.artemis.api.schema.SignedBeaconBlock;
+import tech.pegasys.artemis.provider.JsonProvider;
+
+public class PostBlock implements Handler {
+  public static final String ROUTE = "/validator/block";
+
+  private final JsonProvider jsonProvider;
+  private final ValidatorDataProvider validatorDataProvider;
+  private final SyncDataProvider syncDataProvider;
+
+  public PostBlock(
+      final ValidatorDataProvider validatorDataProvider,
+      final SyncDataProvider syncDataProvider,
+      final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.validatorDataProvider = validatorDataProvider;
+    this.syncDataProvider = syncDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.POST,
+      summary = "Submit a signed transaction to be imported.",
+      tags = {TAG_VALIDATOR},
+      requestBody =
+          @OpenApiRequestBody(content = {@OpenApiContent(from = SignedBeaconBlock.class)}),
+      description =
+          "Submit a signed beacon block to the beacon node to be imported."
+              + " Validation is expected to be performed by the beacon node.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            description = "Block has been successfully broadcast, validated and imported."),
+        @OpenApiResponse(
+            status = RES_ACCEPTED,
+            description =
+                "Block has been successfully broadcast, but failed validation and has not been imported."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Unable to parse request body."),
+        @OpenApiResponse(
+            status = RES_INTERNAL_ERROR,
+            description = "Beacon node experienced an internal error."),
+        @OpenApiResponse(
+            status = RES_SERVICE_UNAVAILABLE,
+            description = "Beacon node is currently syncing.")
+      })
+  @SuppressWarnings("FutureReturnValueIgnored")
+  @Override
+  public void handle(final Context ctx) throws Exception {
+    try {
+      if (syncDataProvider.getSyncStatus().isSyncing()) {
+        ctx.status(SC_SERVICE_UNAVAILABLE);
+        return;
+      }
+
+      final SignedBeaconBlock signedBeaconBlock =
+          jsonProvider.jsonToObject(ctx.body(), SignedBeaconBlock.class);
+
+      validatorDataProvider
+          .submitSignedBlock(signedBeaconBlock)
+          .thenApplyChecked(
+              validatorBlockResult -> {
+                ctx.status(validatorBlockResult.getResponseCode());
+                if (validatorBlockResult.getFailureReason().isPresent()) {
+                  ctx.result(
+                      jsonProvider.objectToJSON(
+                          validatorBlockResult.getFailureReason().get().getMessage()));
+                } else {
+                  ctx.result(jsonProvider.objectToJSON(validatorBlockResult.getHash_tree_root()));
+                }
+                return null;
```

Comment:
is this return actually required? i would have thought it to be redundant..

Reply:
`thenApplyChecked` needs a return object.

Reply:
Please don't add `@SuppressWarnings("FutureReturnValueIgnored")`.  This code is broken as written and will silently ignore any failures because the returned future from `thenApplyChecked` is being ignored.